### PR TITLE
feat: Add autodoc for event annotations

### PR DESCRIPTION
**Pull request by bmtcril**
**Description:** Adds automatic documentation of annotated events for discoverability and ease of configuration. This is in draft until dependencies merge and release, so I can update requirements to the official version, I'll be adding event_key_fields for the rest of the events while that is in review. Generated docs look like this:

Screenshot 2024-02-20 at 1 11 21 PM

**Dependencies:** https://github.com/openedx/code-annotations/pull/98

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Requirements are updated to use the branch of code-annotations necessary to run this so pulling this branch and running `make requirements` should suffice
2. Run `make docs`
3. Sphinx should run and pop open the generated docs in your browser. Navigate to reference -> events to see the content.

**Reviewers:**
- [ ] tag reviewer
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by bmtcril** [comment]
Coverage issue is just because there is more code (comments), there is nothing to test in any of this.

**Review by mariajgrimaldi — COMMENTED** [review]
This has my soft-approval. Thanks for doing this! We'll merge this once https://github.com/openedx/code-annotations/pull/98 is merged so we can the code-annotations release instead. Is that okay?

File: docs/reference/events.rst
Comment (timmc-edx) [inline docs/reference/events.rst:7]: [nit] typo
```suggestion
    Events can be created in other projects and plugins as well, but these default events are guaranteed to exist.
```

**Review by timmc-edx — COMMENTED** [review]
Looks good pending the code-annotations release!

**Comment by bmtcril** [comment]
@timmc-edx @mariajgrimaldi this is ready for re-review. Changes since the last push:

- Using the release of code-annotations
- Ran make upgrade because of that
- Removed pii checks from Makefile and tox since they were broken after the removal of `.pii_annotations.yml`

**Review by mariajgrimaldi — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**
